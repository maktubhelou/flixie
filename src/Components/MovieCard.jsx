import React from 'react'
import PropTypes from 'prop-types'
import CaptionDetail from './CaptionDetail'

const styles = {
  cardStyle: {
    padding: '10px',
    margin: '10px',
    maxWidth: '400px',
    height: '100%',
    backgroundColor: 'rgba(49, 49, 66, 0.8)',
    color: '#eeeeee',
    textAlign: 'center',
  },
  captionWrapper: {
    color: '#888899',
    height: '150px',
    padding: '10px',
    overflowY: 'scroll',
    textAlign: 'justify',
  },
  titleStyle: {
    display: 'inline-block',
    width: '108%',
    minHeight: '48px',
    color: '#bbbbcc',
    paddingTop: '8px',
    paddingBottom: '8px',
    marginTop: '-16px',
    marginBottom: '16px',
    marginLeft: '-16px',
    textAlign: 'center',
    borderBottom: '1px solid #444455',
    backgroundColor: '#111133',
    opacity: '0.9',
    boxShadow: '0px 4px 16px 0px #111133',
  },
  posterStyle: {
    border: '1px solid #444455',
    boxShadow: '0 0 25px 0 #333355',
  },
  headerStyle: {
    paddingLeft: '16px',
  },
  overview: {
    marginBottom: '8px',
    padding: '8px',
    clear: 'both',
  },
}

const MovieCard = ({ movie }) => {
  const {
    id,
    title,
    voteCount,
    popularity,
    score,
    releaseDate,
    overview,
    posterPath,
  } = movie

  const roundedPopularity = Math.floor(popularity)
  const captionDetails = [
    { id: 0, title: 'Release Date', content: releaseDate },
    { id: 1, title: 'Vote average', content: score },
    { id: 2, title: 'Vote count', content: voteCount },
    { id: 3, title: 'Popularity', content: roundedPopularity },
  ]

  return (
    <div style={styles.cardStyle} className="border-rounded">
      <div style={styles.titleStyle}>
        <h1 style={styles.headerStyle}>{title}</h1>
      </div>

      <img
        src={posterPath}
        name={id}
        className="border-rounded"
        style={styles.posterStyle}
        alt=""
      />

      <div style={styles.captionWrapper}>
        {captionDetails.map((detail, i) => (
          <CaptionDetail
            title={detail.title}
            content={detail.content}
            key={detail.id}
            left={i % 2 === 0}
          />
        ))}
        <div style={styles.overview}>{overview}</div>
      </div>
    </div>
  )
}

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    voteCount: PropTypes.number.isRequired,
    popularity: PropTypes.number.isRequired,
    score: PropTypes.number.isRequired,
    releaseDate: PropTypes.string.isRequired,
    overview: PropTypes.string.isRequired,
    posterPath: PropTypes.string.isRequired,
  }).isRequired,
}

export default MovieCard
