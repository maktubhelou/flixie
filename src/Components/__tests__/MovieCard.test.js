import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MovieCard from '../MovieCard'

describe('<MovieCard />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<MovieCard />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<MovieCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
