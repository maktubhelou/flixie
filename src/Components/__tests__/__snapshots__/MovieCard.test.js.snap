// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> matches snapshot 1`] = `
<div
  className="border-rounded"
  style={
    Object {
      "backgroundColor": "rgba(49, 49, 66, 0.8)",
      "color": "#eeeeee",
      "height": "100%",
      "margin": "10px",
      "maxWidth": "400px",
      "padding": "10px",
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#111133",
        "borderBottom": "1px solid #444455",
        "boxShadow": "0px 4px 16px 0px #111133",
        "color": "#bbbbcc",
        "display": "inline-block",
        "marginBottom": "16px",
        "marginLeft": "-16px",
        "marginTop": "-16px",
        "minHeight": "48px",
        "opacity": "0.9",
        "paddingBottom": "8px",
        "paddingTop": "8px",
        "textAlign": "center",
        "width": "108%",
      }
    }
  >
    <h1
      style={
        Object {
          "paddingLeft": "16px",
        }
      }
    />
  </div>
  <img
    alt=""
    className="border-rounded"
    style={
      Object {
        "border": "1px solid #444455",
        "boxShadow": "0 0 25px 0 #333355",
      }
    }
  />
  <div
    style={
      Object {
        "color": "#888899",
        "height": "150px",
        "overflowY": "scroll",
        "padding": "10px",
        "textAlign": "justify",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "#111133",
          "float": "left",
          "marginBottom": "8px",
          "padding": "8px",
        }
      }
    >
      <strong>
        Release date: 
      </strong>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "#111133",
          "float": "right",
          "marginBottom": "8px",
          "padding": "8px",
        }
      }
    >
      <strong>
        Vote average: 
      </strong>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "#111133",
          "float": "left",
          "marginBottom": "8px",
          "padding": "8px",
        }
      }
    >
      <strong>
        Vote Count: 
      </strong>
    </div>
    <div
      style={
        Object {
          "backgroundColor": "#111133",
          "float": "right",
          "marginBottom": "8px",
          "padding": "8px",
        }
      }
    >
      <strong>
        Popularity: 
      </strong>
    </div>
    <div
      style={
        Object {
          "clear": "both",
          "marginBottom": "8px",
          "padding": "8px",
        }
      }
    />
  </div>
</div>
`;
